ui/admin/src/plugins/snackbar.ts

```diff
@@ -0,0 +1,56 @@
+import { App, Plugin, reactive } from "vue";
+
+const SnackbarInjectionKey = Symbol("snackbar");
+
+type SnackbarType = "success" | "error" | "info" | "warning";
+
+interface SnackbarState {
+  message: string;
+  type: SnackbarType;
+  show: boolean;
+}
+
+interface ISnackbarPlugin {
+  showSuccess: (msg: string) => void;
+  showError: (msg: string) => void;
+  showInfo: (msg: string) => void;
+  showWarning: (msg: string) => void;
+  getMessage: () => string;
+  getType: () => SnackbarType;
+  getShow: () => boolean;
```

Comment:
This seems to have a lot of duplication from the regular user interface. Why can we not share the components between the administration and the regular user interface?